src/core/operations/ECDSASignatureConversion.mjs

```diff
@@ -0,0 +1,106 @@
+/**
+ * @author cplussharp
+ * @copyright Crown Copyright 2021
+ * @license Apache-2.0
+ */
+
+import Operation from "../Operation.mjs";
+import r from "jsrsasign";
+
+/**
+ * ECDSA Sign operation
+ */
+class ECDSASignatureConversion extends Operation {
+
+    /**
+     * ECDSASignatureConversion constructor
+     */
+    constructor() {
+        super();
+
+        this.name = "ECDSA Signature Conversion";
+        this.module = "Ciphers";
+        this.description = "Convert an ECDSA signature between hex, asn1 and json.";
+        this.infoURL = "https://wikipedia.org/wiki/Elliptic_Curve_Digital_Signature_Algorithm";
+        this.inputType = "string";
+        this.outputType = "string";
+        this.args = [
+            {
+                name: "Input Format",
+                type: "option",
+                value: [
+                    "Auto",
+                    "ASN.1 HEX",
+                    "Concat HEX",
+                    "JSON"
+                ]
+            },
+            {
+                name: "Output Format",
+                type: "option",
+                value: [
+                    "ASN.1 HEX",
+                    "Concat HEX",
+                    "JSON"
+                ]
+            }
+        ];
+    }
+
+    /**
+     * @param {string} input
+     * @param {Object[]} args
+     * @returns {string}
+     */
+    run(input, args) {
+        let inputFormat = args[0];
+        const outputFormat = args[1];
+
+        // detect input format
+        if (inputFormat === "Auto") {
+            if (input.substr(0, 2) === "30" && r.ASN1HEX.isASN1HEX(input)) {
+                inputFormat = "ASN.1 HEX";
+            } else if (input.indexOf("{") !== -1) {
+                inputFormat = "JSON";
+            } else {
+                inputFormat = "Concat HEX";
+            }
+        }
+
+        // convert input to ASN.1 hex
+        let signatureASN1Hex;
+        switch (inputFormat) {
+            case "ASN.1 HEX":
+                signatureASN1Hex = input;
+                break;
+            case "Concat HEX":
+                signatureASN1Hex = r.KJUR.crypto.ECDSA.concatSigToASN1Sig(input);
+                break;
+            case "JSON": {
+                const inputJson = JSON.parse(input);
+                signatureASN1Hex = r.KJUR.crypto.ECDSA.hexRSSigToASN1Sig(inputJson.r, inputJson.s);
+                break;
+            }
+        }
+
+        // convert ASN.1 hex to output format
+        let result;
+        switch (outputFormat) {
+            case "ASN.1 HEX":
+                result = signatureASN1Hex;
+                break;
+            case "Concat HEX":
```

Comment:
This seems to error out for me, [reproduction example](http://localhost:8080/#recipe=ECDSA_Signature_Conversion('Auto','Concat%20HEX')&input=LS0tLS1CRUdJTiBQVUJMSUMgS0VZLS0tLS0NCk1JR2JNQkFHQnlxR1NNNDlBZ0VHQlN1QkJBQWpBNEdHQUFRQUhoamNZRkNOMVJ2bFBTQUVCdldkWENEM3M4UWENCkVjeDM3dE9rVGEvUDJvaXJURngyUmlZdWhPeWhFMCtNUC9WS2JrUWNvWHVHK0lvRVZ1RDZUNmoxTHNRQTBYQTYNClMrRzNKLytFTmZHVUFVMm42R2R0dkREMWdXaWNEVlR3VmI1ajB6N2xOODd5VTgyREtCOXl5TDllS005dklWTFoNClNrRndvMEVuanJwNXI3Ujloa3M9DQotLS0tLUVORCBQVUJMSUMgS0VZLS0tLS0&ieol=CRLF&oeol=CRLF).

Reply:
You don't convert keys with this but an ECDSA signature. An ECDSA signature consists of two parts an "R" and a "S" point (both are BigInteger). There are different formats how these two values are stored. OpenSSL uses an ASN1 SEQUENCE of two INTEGERs. Windows and JSONWebSigning use a format defined in IEEE-P1363 (just two equally long [zero padded] integer values). I will rename "Concat HEX" to "P1363 HEX" to hopefully make this more clear. I will also add a fourth option "JSONWebSignature", that is the base64url encoded P1363 format used for JWT signatures.

Reply:
Yeah, this is my mistake.  I got very confused between signatures and keys and it made a lot of the comments in this thread inaccurate on my part.  I like the changes you suggest above, and don't believe an error is necessarily incorrect given I've provided entirely the wrong thing!